backend/src/question/question.service.ts

```diff
@@ -0,0 +1,186 @@
+import { Injectable } from "@nestjs/common";
+
+import {
+  FlattenedQuestionContent,
+  FlattenedQuestionSummary,
+  QuestionSummaryFromDb,
+  QUESTION_SUMMARY_SELECT,
+} from "./question.type";
+import { PrismaService } from "../prisma/prisma.service";
+
+@Injectable()
+export class QuestionService {
+  constructor(private prisma: PrismaService) {}
+
+  /**
+   * Gets a summary that matches the unique title slug
+   *
+   * @param   {string} titleSlug  title slug associated to the question
+   *
+   * @return  Corresponding QuestionSummary to the title slug
+   * @throws  NotFoundError
+   */
+  async getContentFromSlug(
+    titleSlug: string
+  ): Promise<FlattenedQuestionContent> {
+    const res = await this.prisma.questionContent.findUniqueOrThrow({
+      where: { titleSlug },
+      select: {
+        content: true,
+        hints: true,
+        summary: {
+          include: {
+            topicTags: true,
+          },
+        },
+      },
+    });
+
+    return {
+      content: res.content,
+      discussionLink: this.getDiscussionLink(titleSlug),
+      hints: res.hints.map((v) => v.hint),
+      topicTags: res.summary.topicTags.map((v) => v.topicSlug),
+    };
+  }
```

Comment:
should include `try` and `catch` for invalid topic slugs and throw the appropriate http exception error, currently returns a 500 error